### PR TITLE
More ModTrees code

### PR DESCRIPTION
**Pull request by Ved-s**
Added TreeLeaf and Shake for ModTrees/ModPalmTrees, fixed smart cursor interaction

### CODE REVIEW

File: patches/tModLoader/Terraria/GameContent/SmartCursorHelper.cs.patch [hunk 1]
```diff
@@ -10,6 +18,85 @@
...
+ 					Tile tile10 = Main.tile[_targets[k].Item1 + l, _targets[k].Item2 + 1];
+ 					if (tile10.active()) {
++						if (TileLoader.CanGrowModTree(tile10.type) || TileLoader.CanGrowModPalmTree(tile10.type))
++							flag = true;
++						else
+-						switch (tile10.type) {
++							switch (tile10.type) {
+-							case 2:
++								case 2:
+-							case 23:
```
Comment (Mirsario) [inline patches/tModLoader/Terraria/GameContent/SmartCursorHelper.cs.patch]: Use gotos and avoid changing indentation in patches.

**Review by Mirsario — CHANGES_REQUESTED** [review]

File: patches/tModLoader/Terraria/GameContent/SmartCursorHelper.cs.patch (hunk 1 above)
Comment (Solxanich) [reply patches/tModLoader/Terraria/GameContent/SmartCursorHelper.cs.patch]: Specific suggestion - Use 
```
flag = true; 
continue;
``` 

And don't touch the switch case

File: patches/tModLoader/Terraria/ModLoader/TileLoader.cs [hunk 2]
```diff
@@ -855,6 +856,18 @@ public static class TileLoader
...
+			var tree = PlantLoader.Get<ModTree>(TileID.Trees, type);
+			if (tree is not null)
+				return tree.CountsAsTreeType;
+
+			var palm = PlantLoader.Get<ModPalmTree>(TileID.PalmTree, type);
+			if (palm is not null)
+				return palm.CountsAsTreeType;
+
+			return TreeTypes.None;
+		}
```
Comment (Solxanich) [inline patches/tModLoader/Terraria/ModLoader/TileLoader.cs]: Move this to PlantLoader

File: patches/tModLoader/Terraria/ModLoader/TileLoader.cs [hunk 3]
```diff
@@ -880,6 +893,18 @@ public static class TileLoader
...
+			var tree = PlantLoader.Get<ModTree>(TileID.Trees, type);
+			if (tree is not null)
+				return tree.Shake(x, y, ref createLeaves);
+
+			var palm = PlantLoader.Get<ModPalmTree>(TileID.PalmTree, type);
+			if (palm is not null)
+				return palm.Shake(x, y, ref createLeaves);
+
+			return true;
+		}
```
Comment (Solxanich) [inline patches/tModLoader/Terraria/ModLoader/TileLoader.cs]: Move this to PlantLoader

File: patches/tModLoader/Terraria/ModLoader/TileLoader.cs [hunk 4]
```diff
@@ -893,10 +918,16 @@ public static class TileLoader
...
-				gore = tree.GrowthFXGore();
+			if (tree is not null)
+				leafGoreType = tree.TreeLeaf();
+			if (leafGoreType > -1)
+				return;
+
+			var palm = PlantLoader.Get<ModPalmTree>(TileID.PalmTree, type);
+			if (palm is not null)
+				leafGoreType = palm.TreeLeaf();
 		}
```
Comment (Solxanich) [inline patches/tModLoader/Terraria/ModLoader/TileLoader.cs]: Move this to PlantLoader

**Review by Solxanich — CHANGES_REQUESTED** [review]
A couple of additional cleanups for relocating methods

**Pull request merged**